### PR TITLE
[Bug]Fix potential overflow of statistics values.

### PR DESCRIPTION
**Pull request by ryuyu**
Add support for stats values up to max value of long.

### CODE REVIEW

File: src/NuGetGallery/ViewModels/UserProfileModel.cs [hunk 1]
```diff
@@ -22,7 +22,7 @@ public UserProfileModel(User user, User currentUser, List<ListPackageItemViewMod
             PackagePage = pageIndex;
             PackagePageSize = pageSize;
 
-            TotalPackageDownloadCount = AllPackages.Sum(p => ((long)p.TotalDownloadCount));
+            TotalPackageDownloadCount = AllPackages.Sum(p => (p.TotalDownloadCount));
```
Comment (joelverhagen) [inline src/NuGetGallery/ViewModels/UserProfileModel.cs:25]:
```suggestion
            TotalPackageDownloadCount = AllPackages.Sum(p => p.TotalDownloadCount);
```

File: src/NuGetGallery/Infrastructure/Lucene/LuceneSearchService.cs [hunk 2]
```diff
@@ -102,8 +102,8 @@ private SearchResults SearchCore(SearchFilter searchFilter)
 
         private static Package PackageFromDoc(Document doc)
         {
-            int downloadCount = Int32.Parse(doc.Get("DownloadCount"), CultureInfo.InvariantCulture);
-            int versionDownloadCount = Int32.Parse(doc.Get("VersionDownloadCount"), CultureInfo.InvariantCulture);
+            long downloadCount = Int64.Parse(doc.Get("DownloadCount"), CultureInfo.InvariantCulture);
```
Comment (joelverhagen) [inline src/NuGetGallery/Infrastructure/Lucene/LuceneSearchService.cs:105]:
```suggestion
            long downloadCount = long.Parse(doc.Get("DownloadCount"), CultureInfo.InvariantCulture);
```

File: src/NuGetGallery/Infrastructure/Lucene/LuceneSearchService.cs [hunk 3]
```diff
@@ -102,8 +102,8 @@ private SearchResults SearchCore(SearchFilter searchFilter)
...
+            long versionDownloadCount = Int64.Parse(doc.Get("VersionDownloadCount"), CultureInfo.InvariantCulture);
```
Comment (joelverhagen) [inline src/NuGetGallery/Infrastructure/Lucene/LuceneSearchService.cs:106]:
```suggestion
            long versionDownloadCount = long.Parse(doc.Get("VersionDownloadCount"), CultureInfo.InvariantCulture);
```

**Comment by joelverhagen** [comment]
Looks like the CI is failing. Could you take a look?

Also, what does the generated SQL look like for the migration `Up` method? You can generate it with `Update-Database -Script`.

**Review by joelverhagen — CHANGES_REQUESTED** [review]
Small comments

**Review by agr — APPROVED** [review]

**Comment by zhhyu** [comment]
Should we consider manually placing an overflowed download count in DEV or INT and test whether anything may be broken (CI, Tests, Manual UI tests)?

**Comment by zhhyu** [comment]
Also, should we have a better understanding about what will happen on DB side? It doesn't seem like a metadata change. Will it cause any on-site issues considering the scale?

**Review by shishirx34 — APPROVED** [review]

**Review by joelverhagen — DISMISSED** [review]

File: tests/NuGetGallery.Facts/ViewModels/UserProfileModelFacts.cs [hunk 4]
```diff
@@ -141,7 +141,7 @@ private ListPackageItemViewModel CreatePackageItemViewModel(string version)
                         DownloadCount = int.MaxValue
                     },
                     Version = version
-                }, currentUser: null);
+                }, currentUser: null);;
```
Comment (dannyjdev) [inline tests/NuGetGallery.Facts/ViewModels/UserProfileModelFacts.cs:144]: nit: remove the last non needed `;`.

**Comment by ryuyu** [comment]
> Also, should we have a better understanding about what will happen on DB side? It doesn't seem like just a metadata change. Will it cause any on-site issues considering the scale?

This is a great concern. After testing this on DEV, I highly recommend we do the PROD transition on a weekend so that the index recreation can proceed at low traffic time (DEV migration, including rebuild, took about 8 minute). Note that there will be downtime between the migration and the deployment of the new gallery. During this time, any requests that hit the Download Count column will probably return 500 as the column is now too wide for the bits. This means we will need to coordinate the release of migration and gallery, and will likely need to leave a status message letting users know that there may be failures during this time while we migrate.

File: tests/NuGetGallery.Facts/ViewModels/UserProfileModelFacts.cs (hunk 4 above)
Comment (ryuyu) [reply tests/NuGetGallery.Facts/ViewModels/UserProfileModelFacts.cs]: Fixed.

**Comment by ryuyu** [comment]
> This migration looks like it might be DTU intensive. We should carefully planned the execution, perhaps at a non-peak time of day.
> 
> @advay26 + @ryuyu - consider holding off the dev to main merge for this PR.

Looks like the code is backwards compatible with either bitwidth in SQL. It should be safe to deploy code and run migration later during down time. Given this, I think it SHOULD be safe to merge, but I'm ok holding off if that is what we want to do @joelverhagen .

**Review by joelverhagen — APPROVED** [review]

**Review by dannyjdev — APPROVED** [review]

**Comment by joelverhagen** [comment]
@ryuyu - is this ready to merge?

**Review by zhhyu — APPROVED** [review]

**Pull request merged**